sdk/src/sys/message.rs

```diff
@@ -1,33 +1,23 @@
 //! Syscalls for reading message metadata.
 
+use fvm_shared::sys::out::message::MessageDetails;
+
 super::fvm_syscalls! {
     module = "message";
 
-    /// Returns the caller's actor ID.
-    ///
-    /// # Errors
-    ///
-    /// None
-    pub fn caller() -> Result<u64>;
-
-    /// Returns the receiver's actor ID (i.e. ourselves).
-    ///
-    /// # Errors
+    /// Returns the details about this invocation.
     ///
-    /// None
-    pub fn receiver() -> Result<u64>;
-
-    /// Returns the method number from the message.
-    ///
-    /// # Errors
-    ///
-    /// None
-    pub fn method_number() -> Result<u64>;
-
-    /// Returns the value that was received.
+    /// - The caller's actor ID.
+    /// - The receiver's actor ID (i.e. ourselves).
+    /// - The method number from the message.
+    /// - The value that was received.
+    /// - The current epoch.
+    /// - The network version.
+    /// - The base fee for the current epoch.
+    /// - The circulating supply.
```

Comment:
Maybe add note that these values are constant during execution and should be cached by sdks.

Reply:
I would also move this documentation to the `MessageDetails` struct itself. Otherwise, it's guaranteed to get out of sync.